internal/core/api.rs

```diff
@@ -337,31 +337,48 @@ impl Window {
 
     /// Returns the position of the window on the screen, in physical screen coordinates and including
     /// a window frame (if present).
-    pub fn position(&self) -> PhysicalPosition {
+    pub fn physical_position(&self) -> PhysicalPosition {
         self.0.window_adapter().position()
     }
 
     /// Sets the position of the window on the screen, in physical screen coordinates and including
     /// a window frame (if present).
     /// Note that on some windowing systems, such as Wayland, this functionality is not available.
-    pub fn set_position(&self, position: PhysicalPosition) {
+    pub fn set_physical_position(&self, position: PhysicalPosition) {
         self.0.window_adapter().set_position(position)
     }
 
+    /// Sets the position of the window on the screen, in physical screen coordinates and including
+    /// a window frame (if present).
+    /// Note that on some windowing systems, such as Wayland, this functionality is not available.
+    pub fn set_logical_position(&self, position: LogicalPosition) {
+        self.0.window_adapter().set_position(position.to_physical(self.scale_factor()));
```

Comment:
I think that this, as well as in the C++ equivalent, will not produce the correct result when the window isn't shown yet - because the scale factor is not known yet. That would break the use-case of saving and restoring the logical size of a window in a configuration file. In the process of restoring it, this setter would be called but it would effectively treat it as physical size.

We might need to use an enum then, even if just internally (WindowAdapterSealed is private).

Reply:
Good point, this needs some extra logic.

This also means that we should expand the FFI interface we expose so that we do not need to do this in C++ again.

Reply:
> We might need to use an enum then, even if just internally (WindowAdapterSealed is private).

Thinking a bit more about this: right now WindowAdapterSealed::set_inner_size is private. But in the future we will need to make it public.  And then we will kind of have to make that enum public.
Is that not an argument in favor of having a Size and Position enum just like winit has?

Reply:
> Is that not an argument in favor of having a Size and Position enum just like winit has?

Yes, you're right.

Reply:
Why an enum? Don't we need both?